quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/DebugHostnameSettingsResource.java

```diff
@@ -117,10 +130,44 @@ public String debug(final @PathParam("realmName") String realmName) throws IOExc
     @GET
     @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX + "/" + PATH_FOR_TEST_CORS_IN_HEADERS)
     @Produces(MediaType.TEXT_PLAIN)
-    public Response test(final @PathParam("realmName") String realmName) {
-        Response.ResponseBuilder builder = Response.ok(PATH_FOR_TEST_CORS_IN_HEADERS + "-OK");
-        String origin = keycloakSession.getContext().getRequestHeaders().getHeaderString(Cors.ORIGIN_HEADER);
-        builder.header(Cors.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+    public Response test(final @PathParam("realmName") String realmName, @DefaultValue("false") @QueryParam("frontEnd") boolean frontEnd) {
+        String text = "OK";
+        String corsOrigin = keycloakSession.getContext().getRequestHeaders().getHeaderString(Cors.ORIGIN_HEADER);
+        URI requestUri = keycloakSession.getContext().getUri().getRequestUri();
```

Comment:
I did some testing and this actually does get us the "raw" request but one enriched by the Hostname provider, see:
https://github.com/keycloak/keycloak/blob/8d9439913cffe5aef10ac14eb2876daca3336e5e/server-spi/src/main/java/org/keycloak/models/KeycloakUriInfo.java#L52

@shawkins Could you please check it?

Reply:
I have. You only get the enrichment if you call getBaseUri. If you call getRequestUri, you get the underlying request: https://github.com/keycloak/keycloak/blob/8d9439913cffe5aef10ac14eb2876daca3336e5e/server-spi/src/main/java/org/keycloak/models/KeycloakUriInfo.java#L72

Reply:
If you have hostname=example.com the urls tested by hostname-debug will be to example.com, not localhost.

Reply:
> If you have hostname=example.com the urls tested by hostname-debug will be to example.com, not localhost.

Right, sorry, I was stupid and also forgot I have `example.com` in my `/etc/hosts`. :D

LGTM, thanks. Let's get this merged finally.

Reply:
Just to double check - do you think we need to review our usage of getRequestUri and the other methods on KeycloakUriInfo that expose the underlying delegate? I see several code paths using that, which makes using the full hostname url without proxy-headers problematic - or are we good with just requiring the proxy-headers to be be set?

Reply:
> or are we good with just requiring the proxy-headers to be be set

IMHO we're good for now - we say in docs proxy headers are required anyway for edge and reencrypt. If we did a refactor of `getRequestUri`, I would do that in a separate PR. But I wouldn't do that for now not to risk regressions.